Satellite-im/Core-PWA
# test -add jest tests + pre commit hook

**What this PR does** 📖
- Add Jest test for `Crypto.ts`
- Add Jest test for `Security.ts`
- Adds `husky` and `lint-staged`
- Adds pre-commit hook which handles linting before performing a commit
- Adds `vue` linter
- Run linting on `.vue` files



note - when you add a commit, the pre-commit lint will run only on the staged files, the ones that were modified

- in case it passes the linter, goes to the commit message editor and you can commit
- in case it fails the linter, aborts but all your changed files are still the same locally

**Special notes for reviewers** 🗒️
- To run the tests, use yarn test

Tests will appear as follows

<img width="1104" alt="Captura de ecrã 2021-11-24, às 21 08 00" src="https://user-images.githubusercontent.com/29093946/143312991-8f162981-53b6-42f9-a30e-00e23b671010.png">

To test the pre-commit hook, edit any files with something such as ''testinglinter;;;;''

<img width="609" alt="Captura de ecrã 2021-11-25, às 00 40 02" src="https://user-images.githubusercontent.com/29093946/143330204-391771a4-d477-428d-947b-349e7baefcc8.png">

Will run the tests and run the linter


**Additional comments** 🎤

- Verify linted files
- Verify linter rules
